components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -377,19 +377,19 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       let notifications: Record<number, RewardsExtension.Notification> = state.notifications
       let id = ''
 
-      if (!notifications) {
-        notifications = []
+      if (!notifications || Array.isArray(notifications)) {
+        notifications = {}
```

Comment:
This is the main fix. @NejcZdovc can you explain better about:
- why this was an array
- what the repeatable case could be to get in this state where either notification is blank or existing state.notifications is an array (which was my case - `state.notifications` was already `[]`) so that we can get a test plan

Reply:
@petemill line 155 in this same file seems to set it, line 168 merges it back into state.

Reply:
I think that I copied it from line 156 where we set it to array and I guess that was problematic as well. So I think we need to fix line 156 as well. You can see that in tests that I added it was expected to be an object https://github.com/brave/brave-core/pull/2296/commits/9cecb152a1056547664dada7c6de900722fafe4f#diff-f250acb8ae40c4d514754045d9b6e936R379.

Not sure as why only few people see this problem as it should break for everyone

Reply:
@bbondy @NejcZdovc that in L155 also needs to change, right? It's invalid to create this property as an array when it's used as an object.

Reply:
@NejcZdovc would only be a problem if `state.notifications` does not yet exist, right? Do we persist this f/e redux state to disk? Me, personally, I 'upgraded' a profile from dev to nightly, if that helps

Reply:
The type in L153 also appears to be incorrect, and assume that notification ID is always an integer: 
```
let notifications: Record<number, RewardsExtension.Notification> = state.notifications
```

Reply:
yup we need to change line 155. It's strange why it would work for some and not work for others. Also we had this code (line 155) in there for ages

Reply:
I can't remember the specific details, but wondering if this is fallout from the notifications type change. I seem to recall they were initially integers and then switched to strings. I wonder if that caused a problem and/or wasn't fully refactored...